pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -373,7 +376,8 @@ def determine_ddp_device_ids(self):
 
     def pre_dispatch(self):
         # share ddp pids to all processes
-        self._share_information_to_prevent_deadlock()
+        if self._should_run_deadlock_detection():
```

Comment:
there is the problem now that rank 0 will enter this if statement and rank > 0 will not, then hang later on on the broadcast of `share_information_..`

Reply:
:facepalm

Reply:
great, these things now run for me successfully using ddp with this branch: 

- `tests/accelerators/test_ddp.py`
- `PL_RUNNING_SPECIAL_TESTS=1 PL_RECONCILE_PROCESS=1 python tests/plugins/environments/torch_elastic_deadlock.py`
- tested on another model in my projects

Reply:
hopefully CI agrees